projects/core/components/primitive-textfield/test/primitive-textfield.component.spec.ts

```diff
@@ -1,97 +1,52 @@
-import {Component, ViewChild} from '@angular/core';
+import {HarnessLoader} from '@angular/cdk/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {Component} from '@angular/core';
 import {ComponentFixture, TestBed} from '@angular/core/testing';
-import {By} from '@angular/platform-browser';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {TuiRootModule} from '@taiga-ui/core/components/root';
-import {
-    configureTestSuite,
-    TuiNativeInputPO,
-    TuiPageObject,
-    tuiTestCleaner,
-    tuiTestPlaceholder,
-    tuiTestTooltip,
-} from '@taiga-ui/testing';
+import {configureTestSuite, TuiPrimitiveTextfieldHarness} from '@taiga-ui/testing';
 
 import {TuiHintModule} from '../../../directives/hint/hint.module';
 import {TuiTextfieldControllerModule} from '../../../directives/textfield-controller';
-import {TuiSizeL, TuiSizeS} from '../../../types/size';
-import {TuiPrimitiveTextfieldComponent} from '../primitive-textfield.component';
 import {TuiPrimitiveTextfieldModule} from '../primitive-textfield.module';
 
 describe(`PrimitiveTextfield`, () => {
     @Component({
         template: `
+            <tui-primitive-textfield id="test1"></tui-primitive-textfield>
```

Comment:
can we move their down into different describes without ids?

Reply:
sorry, I dont get it. Do you mean to replace by more descriptive classes ?

Reply:
sorry for my English)

I mean

```ts
describe(`PrimitiveTextfield`, () => {
    describe(`case 1`, {
       @Component({
        template: `<tui-primitive-textfield></tui-primitive-textfield>`,
        // ...
        })

        it(`...`, () => {
            const component = await loader.getHarness(TuiPrimitiveTextfieldHarness);
        });
    });


    describe(`case 2`, {
       @Component({
        template: `
            <tui-primitive-textfield
                [pseudoFocus]="focused"
                [value]="''"
            >
                <input
                    tuiTextfield
                    [attr.placeholder]="'placeholder'"
                />
            </tui-primitive-textfield>
         ,
        // ...
        })

        it(`...`, () => {
            const component = await loader.getHarness(TuiPrimitiveTextfieldHarness);
        });
    });

});

```

Reply:
it seems to me that it is much clearer what we are testing in a particular case

Reply:
no worries, yes agreed, this is much nicer. will do the refactor.

Reply:
I think generally test files will get longer then, because we will have to define`ConfigureTestSuit` in each describe. For example 

```
describe(`PrimitiveTextfield 3`, () => {
    @Component({
        template: `
            <tui-primitive-textfield
                id="test2"
                [filler]="'filler'"
                [postfix]="'post'"
                [value]="'value'"
                [pseudoFocus]="true"
            ></tui-primitive-textfield>
        `,
    })
    class TestComponent {}

    let fixture: ComponentFixture<TestComponent>;
    let loader: HarnessLoader;

    configureTestSuite(() => {
        TestBed.configureTestingModule({
            imports: [
                NoopAnimationsModule,
                TuiPrimitiveTextfieldModule,
                TuiTextfieldControllerModule,
                TuiHintModule,
                TuiRootModule,
            ],
            declarations: [TestComponent],
        });
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(TestComponent);
        loader = TestbedHarnessEnvironment.loader(fixture);
    });

    it(`is not shown when value is empty and field is not focused`, async () => {
        const component = await loader.getHarness(
            TuiPrimitiveTextfieldHarness.with({selector: `#test1`}),
        );
        const valueDecorationText = await component.valueDecorationText();

        expect(valueDecorationText).toBe(``);
    });
});


```

I like the way material components are doing. 
https://github.com/angular/components/blob/main/src/material/legacy-button/testing/shared.spec.ts